File: src/main/java/com.raworkstudio.spring/CustomPromptProvider.java
package com.raworkstudio.spring;

import org.springframework.core.Ordered;
import org.springframework.core.annotation.Order;
import org.springframework.shell.plugin.support.DefaultPromptProvider;
import org.springframework.stereotype.Component;

/**
 * Created by Ivan Alburquerque on 3/27/2017.
 */
@Component
@Order(Ordered.HIGHEST_PRECEDENCE)
public class CustomPromptProvider extends DefaultPromptProvider {

    @Override
    public String getPrompt() {
        return "mvnsh-shell>";
    }


    @Override
    public String getProviderName() {
        return "My prompt provider";
    }

}

File: src/main/kotlin/com.raworkstudio.spring/configurations/CustomPromptProvider.kt
package com.raworkstudio.spring.configurations

import org.springframework.core.Ordered
import org.springframework.core.annotation.Order
import org.springframework.shell.plugin.support.DefaultPromptProvider
import org.springframework.stereotype.Component

/**
 * Created by Ivan Alburquerque on 3/31/2017.
 */
@Component
@Order(Ordered.HIGHEST_PRECEDENCE)
open class CustomPromptProvider : DefaultPromptProvider() {

    override fun getProviderName(): String {
        return "My prompt provider"
    }

    override fun getPrompt(): String {
        return "mvn-shell>"
    }
}